app/js/lib/form_components.js

```diff
@@ -42,7 +42,10 @@ var objectAssign = require( 'object-assign' ),
 		onChange: null,
 		validator: null,
 		render: function ( formContent ) {
-			this.element.val( formContent[ this.contentName ] );
+			if ( this.element.val() !== formContent[ this.contentName ] ) {
+				this.element.val( formContent[ this.contentName ] );
+				this.element.change();
+			}
```

Comment:
While I agree that triggering a DOM event during rendering feels somewhat icky, it's the correct behavior none the less. Well done.